### ISSUE TITLE
Question about loss computation and gradients in the example.

### ISSUE THREAD

**Issue by li012589** [nicola-decao/s-vae-pytorch#15]
Hi, I have a question about the loss computation in the MNIST example. 

According to your paper, the gradients should have two terms, one named g_rep and the other named g_cor. The g_cor is computed using log derivative trick which is like the reinforce/score function estimator. But in [L151](https://github.com/nicola-decao/s-vae-pytorch/blob/master/examples/mnist.py#L151C40-L151C40), I think only the g_rep term is considered. Did I miss something? Shouldn't L151 be like 
`loss = loss_recon + loss_recon.detach() * log_g_div_r(eps, ) + loss_KL`

**Comment by trdavidson** [comment]
Hi there, 

Thanks for reaching out and for your interest in our work!

You are correct: During experiments we found computing the `g_cor` term to be problematic at times and that dropping it did not negatively impact results. Theoretically, it would of course be more correct to include the term.

Cheers,
-Tim

**Comment by li012589** [comment]
@trdavidson Thanks for the reply! I also have a following questions regarding the equations of the g_cor term. In your paper, this term  is derived from g/r where g is the von Mises-Fisher PDF and r is the distribution from the invertible transformation h. Then, you have the relatively complex equation in the App., which is Eq. (39). I wonder if there may be a simpler form of this equation. In the App. of your paper, the until condition from Alg. 2 may be a good candidate. This appears in the same position as the q/r term in Alg. 1 from Naesseth2017. Also, this is exactly the probability pi( u | epsilon ) in the joint distribution pi( u, epsilon ). Then, when you integrate out the u variable as in Eq. (4) of Naesseth2017. This term also appears in the corresponding position.
